consensus-types/light-client/update.go

```diff
@@ -578,15 +685,29 @@ func (u *updateElectra) SetNextSyncCommitteeBranchElectra(branch [][]byte) error
 		return err
 	}
 	u.nextSyncCommitteeBranch = b
+
+	u.p.NextSyncCommitteeBranch = branch
+
 	return nil
 }
 
 func (u *updateElectra) FinalizedHeader() interfaces.LightClientHeader {
 	return u.finalizedHeader
 }
 
-func (u *updateElectra) SetFinalizedHeader(header interfaces.LightClientHeader) {
+func (u *updateElectra) SetFinalizedHeader(header interfaces.LightClientHeader) error {
+	if header.Version() != version.Electra {
+		return fmt.Errorf("header version %s is not %s", version.String(header.Version()), version.String(version.Electra))
+	}
 	u.finalizedHeader = header
+
+	proto, ok := header.Proto().(*pb.LightClientHeaderDeneb)
+	if !ok {
+		return fmt.Errorf("header type %T is not %T", proto, &pb.LightClientHeaderDeneb{})
+	}
```

Comment:
Electra update uses Deneb header because there is no change to the header in Electra fork